amanuensis/blueprints/download_urls.py

```diff
@@ -40,55 +39,36 @@ def download_data(project_id):
     if not project_id:
         raise UserError("A project_id is needed to retrieve the correct URL")
 
-    project = get_by_id(logged_user_id, project_id)
-    if not project:
-        raise NotFound("The project with id {} has not been found.".format(project_id))
-
-    associated_users_ids = []
-    associated_users_emails = []
-    for associated_user_role in project.associated_users_roles:
-        if associated_user_role.active and associated_user_role.associated_user.user_id and associated_user_role.role.code == "DATA_ACCESS":
-            associated_users_ids.append(associated_user_role.associated_user.user_id)
-        if associated_user_role.active and associated_user_role.associated_user.email and associated_user_role.role.code == "DATA_ACCESS":
```

Comment:
I forgot, what is the assumption in not checking the present of one of these email and user_id anymore?

Reply:
"user = get_project_associated_users(session, project_id, associated_user_user_id=logged_user_id, associated_user_email=logged_user_email, many=False, throw_not_found=True)" this part in the new version searches the project_has_associated_user table for a row matching the project_id, logged in user_id and email so if that isnt present in the the table a Not_Found will be thrown so the check is just handled down a layer now